config/deploy/docker/lib/aws_sdk_helpers.rb

```diff
@@ -0,0 +1,105 @@
+    define_method(:autoscaling)      { AwsSdkHelpers::ClientMethods.autoscaling }
+    define_method(:ec2)              { AwsSdkHelpers::ClientMethods.ec2 }
+    define_method(:cloudwatchevents) { AwsSdkHelpers::ClientMethods.cloudwatchevents }
+    define_method(:cw)               { AwsSdkHelpers::ClientMethods.cw }
+    define_method(:cwl)              { AwsSdkHelpers::ClientMethods.cwl(
+      profile: respond_to?(:_cluster_name) ? _cluster_name : AwsSdkHelpers::Helpers.cluster_name,
+    ) }
+  end
+
+  module Helpers
+    include AwsSdkHelpers::ClientMethods
+
+    def self.cluster_name
+      ENV.fetch('CLUSTER_NAME', ENV.fetch('AWS_CLUSTER', ENV.fetch('AWS_PROFILE', ENV.fetch('AWS_VAULT', ''))))
+    end
+    def _cluster_name
+      AwsSdkHelpers::Helpers.cluster_name
+    end
+
+    def self.capacity_providers(cluster)
+      r = {}
+      capacity_provider_names = AwsSdkHelpers::ClientMethods.ecs.describe_clusters(clusters: [cluster]).clusters.first.capacity_providers
+      AwsSdkHelpers::ClientMethods.ecs.describe_capacity_providers(capacity_providers: capacity_provider_names).capacity_providers.map do |capacity_provider|
+        asg_name = capacity_provider.auto_scaling_group_provider.auto_scaling_group_arn.split('/').last
+        asg = AwsSdkHelpers::ClientMethods.autoscaling.describe_auto_scaling_groups(auto_scaling_group_names: [ asg_name ]).auto_scaling_groups.first
+
+        launch_template_id = asg.mixed_instances_policy.launch_template.launch_template_specification.launch_template_id
+        launch_template_versions = AwsSdkHelpers::ClientMethods.ec2.describe_launch_template_versions(launch_template_id: launch_template_id)
+
+        ami_id = launch_template_versions.launch_template_versions[0].launch_template_data.image_id
+
+        r[capacity_provider.name] = {
+          name: capacity_provider.name,
+          ami_id: ami_id
+        }
+      end
+      return r
+    end
+    def _capacity_providers(cluster = nil)
+      cluster ||= _cluster_name
+      @capacity_providers ||= AwsSdkHelpers::Helpers.capacity_providers(cluster)
+    end
+
+    def self.default_placement_constraints(ami_id:)
+      [
+        {
+          expression: "attribute:ecs.ami-id == #{ami_id}",
+          type: 'memberOf'
+        },
+      ]
+    end
+    def _default_placement_constraints(capacity_provider_name: '', ami_id: '')
+      if ami_id.empty?
+        unless capacity_provider_name.empty?
+          ami_id = _capacity_providers[capacity_provider_name][:ami_id]
+        else
+          ami_id = _capacity_providers[_capacity_providers.keys.first][:ami_id]
+        end
+      end
+      AwsSdkHelpers::Helpers.default_placement_constraints(ami_id: ami_id)
+    end
+
+
+    def _spot_capacity_provider_name
+      _capacity_providers.find { |k,v| v[:name].match(/spt-v2/) }[1][:name]
+    end
+
+    def _on_demand_capacity_provider_name
+      _capacity_providers.find { |k,v| v[:name].match(/ondemand-v2/) }[1][:name]
```

Comment:
Should these pull the `v2` out of an environment variable?  Seems like one more place we'll need to keep track of a version.  I'd rather see it in the environment, or at least in a constant.

Reply:
I like that @eanders. Let's not block this PR, but I made https://www.pivotaltracker.com/story/show/181282309.